Makefile

```diff
@@ -8,268 +28,405 @@ usage:
+ci-job-libraries:
+	$(call banner,CI-Job: Libraries)
+	@cd libraries/enum_primitive && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/riscv-csr && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/tock-cells && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/tock-register-interface && CI=true RUSTFLAGS="-D warnings" cargo test
+	@cd libraries/tock-rt0 && CI=true RUSTFLAGS="-D warnings" cargo test
 
-.PHONY: audit
-audit:
-	@for f in `./tools/list_lock.sh`; do echo "$$(tput bold)Auditing $$f"; (cd "$$f" && cargo audit || exit 1); done
+.PHONY: ci-job-archs
+ci-job-archs:
+	$(call banner,CI-Job: Archs)
+	@for arch in `./tools/list_archs.sh`;\
+		do echo "$$(tput bold)Test $$arch";\
+		cd arch/$$arch;\
+		CI=true RUSTFLAGS="-D warnings" TOCK_KERNEL_VERSION=ci_test cargo test || exit 1;\
+		cd ../..;\
+		done
+
+.PHONY: ci-job-kernel
+ci-job-kernel:
+	$(call banner,CI-Job: Kernel)
+	@cd kernel && CI=true RUSTFLAGS="-D warnings" TOCK_KERNEL_VERSION=ci_test cargo test
 
-.PHONY: emulation-setup
-emulation-setup: SHELL:=/usr/bin/env bash
-emulation-setup:
-	@#Use the latest QEMU as it has OpenTitan support
-	@printf "Buildling QEMU, this could take a few minutes\n\n"
+.PHONY: ci-job-chips
+ci-job-chips:
+	$(call banner,CI-Job: Chips)
+	@for chip in `./tools/list_chips.sh`;\
+		do echo "$$(tput bold)Test $$chip";\
+		cd chips/$$chip;\
+		CI=true RUSTFLAGS="-D warnings" TOCK_KERNEL_VERSION=ci_test cargo test || exit 1;\
+		cd ../..;\
+		done
+
+.PHONY: ci-setup-tools
+ci-setup-tools:
+ifdef CI
+ifeq ($(PLATFORM),Linux)
+	sudo apt-get install libusb-1.0.0-dev
+else ifeq ($(PLATFORM),Darwin)
+	brew install libusb-compat pkg-config
+else
+	$(error CI on unsupported platform.)
+endif
+else
+	@pkg-config --cflags --libs libusb > /dev/null || (\
+		printf "\n$$(tput bold)Missing Dependency$$(tput sgr0)";\
+		printf "\n";\
+		printf "Need to install 'libusb' for development on your system.\n";\
+		printf "  - Debian: sudo apt-get install libusb-1.0.0-dev\n";\
+		printf "  - Darwin: brew install libusb-compat pkg-config\n";\
+		exit 1)
+endif
+
+.PHONY: ci-job-tools
+ci-job-tools: ci-setup-tools
+	$(call banner,CI-Job: Tools)
+	@for tool in `./tools/list_tools.sh`;\
+		do echo "$$(tput bold)Build & Test $$tool";\
+		cd tools/$$tool;\
+		CI=true RUSTFLAGS="-D warnings" cargo build --all-targets || exit 1;\
+		cd - > /dev/null;\
+		done
+
+
+
+### ci-runner-github-qemu jobs:
+
+# ci-setup-qemu uses make as intended a bit to get all the needed parts
+.PHONY: ci-setup-qemu
+ci-setup-qemu: tools/qemu/riscv32-softmmu tools/qemu-runner/opentitan-boot-rom.elf
+
+tools/qemu/riscv32-softmmu:
+	@# Use the latest QEMU as it has OpenTitan support
+	@printf "Building QEMU, this could take a few minutes\n\n"
+	# Download Tock qemu fork if needed
 	@if [[ ! -d tools/qemu || ! -f tools/qemu/VERSION ]]; then \
 		rm -rf tools/qemu; \
 		cd tools; git clone https://github.com/alistair23/qemu.git --depth 1 -b riscv-tock.next; \
```

Comment:
I think it's fine to leave it like this in the scope of this pull request, but let's not forget to add some kind of integrity check from Tock's side, i.e. cloning at a specific commit vetted by Tock (or checking that HEAD is at a specific commit).